app/src/main/java/com/android/streetworkapp/model/progression/Progression.kt

```diff
@@ -33,7 +33,7 @@ data class Achievement(
     val icon: Int,
     val title: String,
     val tags: List<String>,
-    val description: String
+    var description: String
```

Comment:
Why does it have to be a variable? Can the description change after initialization?

Reply:
Yes, it is used in the exercises achievements in the UI, to display the record of the user. We hope that the user will do a better record, and thus we will need to update the description.

Reply:
But doesn't updating the ViewModel overwrite the current progress with a new object every time an update is made in the database, making the variable concept useless?

I'll let you have a look at the User ViewModel for example, the User data class has only values and not variables - every repository operation update the `_user` value with a new value.

What do you think of it? It's a detail, so if you don't want to change, no problem, but did you understand what I was talking about?

Reply:
I think you are confused between this enum and the attributes of the ViewModel. A new attribute will probably be added, similar to the _currentUser you are talking about, with a mutableStateFlow etc... 
Here, this enum is not a mutable state flow, it is juste a way to store information about Achievements. (Notice that the progression collection in the  database does not store any achievement, only a "name" of this achievement, that is then used to display the correct element of the enum)
If it is still not clear, we can discuss it in person as it is quite complicated by messages.

Reply:
We talked about it in person - I hadn't quite understood how achievements worked, but now it's fine and we'll leave it that way.